secretstores/hashicorp/vault/vault.go

```diff
@@ -96,6 +110,16 @@ func (v *vaultSecretStore) Init(metadata secretstores.Metadata) error {
 
 	v.vaultAddress = address
 
+	v.vaultEnginePath = defaultVaultEnginePath
+	if val, ok := props[vaultEnginePath]; ok && val != "" {
+		v.vaultEnginePath = val
+	}
+
+	v.parseAsMap = true
```

Comment:
instead of parseAsMap what about the opposite and set something like "valueAsString". Another way, would be more extensible: "valueType" and then accept enum values like "map" and "text".

Reply:
Good! Fixing all above by refactoring parseAsMap to vaultValueType, which accept two values map and text.